### PR TITLE
Allow users to supply a Slack app id and secret.

### PR DESCRIPTION
**Pull request by jonthegeek**
I'm still getting tests sorted out, but this should work. This way users can authenticate using their own Slack apps, and thus can use this for logging into their own sites (with their own redirect URLs; right now we're restricted to localhost or the r4ds shinyapps account).

### CODE REVIEW

File: R/team_managment.R [hunk 1]
```diff
@@ -35,8 +36,13 @@ add_team <- function(team, token) {
...
 #' @rdname manage_team
 #' @export
-add_team_token <- function(team, token, verbose = TRUE) {
-
+add_team_token <- function(team,
+                           token,
+                           verbose = TRUE,
+                           client_id = NULL,
+                           client_secret = NULL) {
+  client_id <- client_id %||% builtin_client_id
```
Comment (yonicd) [inline R/team_managment.R]: why is this needed here? it isnt used in the function. same line 45

Comment (yonicd) [reply R/team_managment.R]: cleaned up in new commit

**Comment by jonthegeek** [comment]
Converted back to draft. I want to add a vignette walking through the process of supplying your own client (and walk through that process myself to make sure it works as expected; it's been a long time since I created a new client).

**Review by yonicd — APPROVED** [review]

**Comment by jonthegeek** [comment]
Tweaking the vignette.

**Comment by yonicd** [comment]
not sure why i thought formal in `Sys.getenv` was `default`

**Review by yonicd — APPROVED** [review]

**Comment by jonthegeek** [comment]
> not sure why i thought formal in `Sys.getenv` was `default`

You had them as getOption, which *does* use `default`. I switched it to `Sys.getenv`, because it's easier/more likely for a user to set these things through an environment variable (eg, `.Renviron`, which shinyapps.io recognizes) rather than an option.
